src/App.js

```diff
@@ -14,11 +17,94 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    wasStarted: false,
+    wordLength: 1,
+  }
+
+  start = () => this.setState({ wasStarted: true });
+
+  reset = () => this.setState({
+    goods: goodsFromServer,
+    wordLength: 1,
+  });
+
+  reverseList = () => (
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }))
+  );
+
+  sortAlphabetically = () => (
+    this.setState(state => ({
+      goods: [...state.goods].sort(),
+    }))
+  )
+
+  sortByLength = () => (
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }))
+  )
+
+  selectByLength = event => this.setState({ wordLength: event.target.value });
+
+  render() {
+    const { wasStarted, goods, wordLength } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__header">Goods</h1>
+
+        <div>
+          <Button
+            wasStarted={!wasStarted}
+            call={this.reset}
```

Comment:
what does call mean here?

Reply:
function call. i changed this to action. is it better now?